### DISCUSSION TITLE
Will pymupdf support `PAX` attribution?

### DISCUSSION THREAD

**Discussion by xsank** [pymupdf/PyMuPDF#2941] [Q&A]
Some paper use PAX as `cite`
eg:
```shell
α∗
th
{'kind': 4, 'xref': 314, 'from': Rect(144.49099731445312, 105.655029296875, 148.4770050048828, 123.5889892578125), 'nameddest': 'PAX%40.%2Fpapers%2F224_Paper.pax%401', 'id': ''}
e1
{'kind': 4, 'xref': 316, 'from': Rect(249.94500732421875, 505.87701416015625, 257.9150085449219, 520.0850219726562), 'nameddest': 'PAX%40.%2Fpapers%2F224_Paper.pax%402', 'id': ''}
(Head
r sum-
{'kind': 4, 'xref': 317, 'from': Rect(264.44500732421875, 623.175048828125, 291.1260070800781, 636.9190063476562), 'nameddest': 'PAX%40.%2Fpapers%2F224_Paper.pax%403', 'id': ''}
```

Pymupdf could find the `from` rect, but missing the `to` point

I could not find any related docs in the website：https://pymupdf.readthedocs.io/en/latest/search.html?q=PAX&check_keywords=yes&area=default

will pymupdf support this feature? thanks

**Comment by JorjMcKie** [comment]
This is already supported!
You are showing so-called "named" destination links. They do not have a "to" attribute because the symbolic name given under "nameddest" has all that information.
Please use a variant of the import statement `import fitz_new as fitz` to activate full named destination support.
This "rebased" implementation will replace the current, "classic" implementation during the first half of January.

**Comment by xsank** [comment]
nameddest only contains the information of `PAX@./papers/224_Paper.pax@5`

**Comment by xsank** [comment]
In fact, `cite` have the `to` attribution,

**Comment by JorjMcKie** [comment]
> In fact, `cite` have the `to` attribution,

What is cite?

**Reply by JorjMcKie** [reply]
Or are you referring to the HTML tag `<cite>`?
If yes:
We are talking of PDFs here - there is not such thing in PDF. What you seem to have in your PDF are **_named stinations_**. And those are supported by PyMuPDF as mentioned.

**Comment by JorjMcKie** [comment]
I am going to move this to the "Discussions" tab. This is no issue.

**Comment by xsank** [comment]
```shell
 an
{'kind': 4, 'xref': 6, 'from': Rect(229.4770050048828, 636.8400268554688, 235.45399475097656, 649.3610229492188), 'page': 0, 'to': Point(87.006, 80.829), 'zoom': 0.0, 'nameddest': 'Hfootnote.3', 'id': ''}
(Sinha et al.,
for scholarly
{'kind': 4, 'xref': 7, 'from': Rect(445.02099609375, 501.156005859375, 493.64898681640625, 512.0999755859375), 'page': 7, 'to': Point(301.161, 635.532), 'zoom': 0.0, 'nameddest': 'cite.Sinha2015AnOO', 'id': ''}
 2015)
data in
{'kind': 4, 'xref': 8, 'from': Rect(497.67999267578125, 501.156005859375, 519.5980224609375, 512.0999755859375), 'page': 7, 'to': Point(301.161, 635.532), 'zoom': 0.0, 'nameddest': 'cite.Sinha2015AnOO', 'id': ''}
```
like this. the `cite`, `footnote`, `PAX` are all PDF attributions. so that the pdf reader could jump the specific position of the page.

**Reply by JorjMcKie** [reply]
You obviously used `import fitz_new` this time.
If you extract the page links via `page.get_links()` in this implementation, then indeed the method already does the destination resolution for your comfort, so you will find "to", "page" and other attributes.

But you still misunderstand: the words in the name, like "footnote", "cite" have no meaning in PDF! It is just an identifying string - not more. Could also be "abracadabra123_xyz" without any adverse effect.

So I hope you will accept that your initial requirement is already satisfied - no enhancement necessary.

**Comment by xsank** [comment]
`cite`, `footnote` have the `to` point, but `PAX` not.

**Reply by JorjMcKie** [reply]
Well, if you still have concerns then please attach that problem PDF here.

**Reply by JorjMcKie** [reply]
Let me try another explanation, because you seem to be unaware of what PDF offers as hyperlink kinds:
* GoTo:  jump to another page inside the same document
* GoToR:  jump to some page in another document
* Launch:  start an application or open a document (no jumping to a particular point)
* URI:  jump to a resource given in HTML syntax. Can be again a file or an internet resource, etc.)
* Named:  jump to a named destination. Details are encoded in the definition of the symbolic name.

All of the above are supported by PyMuPDF.

* GoToE: similar to GoToR, but the document is contained as an attachment inside the current document. This is **_not supported_** by PyMuPDF.

There is nothing outside the above list which can be used as a hyperlink in PDF.
Again: things like "cite", "quote", "footnote" are undefined in PDF. The same is true for "PAX" ... whatever that may be.

**Comment by xsank** [comment]
this discussions could be closed, i found the full points in the doc.resolve_names(), thanks again finally.
